From b0b004c92b128f07e3cbe82fe299610cde3fabd5 Mon Sep 17 00:00:00 2001
From: imdtom <thomas.bourlard@imd-tec.com>
Date: Thu, 14 Aug 2025 14:24:49 +0100
Subject: [PATCH] Added DT node to enable SSC Sensors (#15) (#36)

Added DT node to enable kernel module
which will load the adsp firmware and provide
a way to boot the ADSP/SLPI (Sensors Low Power Island).

This is actually a workaround because Qualcomm's actual adsp-loader
kernel module is in the audio-kernel drivers. However, we have not
included the audio-kernel drivers into our image yet.
The SSC kernel driver works very similarly to the adsp-loader
and actually works for our usecase without pulling all the audio-kernel drivers.
---
 qcom/kalamap-imdt-sbc8550.dtsi | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/qcom/kalamap-imdt-sbc8550.dtsi b/qcom/kalamap-imdt-sbc8550.dtsi
index 78d2e46..c77a086 100644
--- a/qcom/kalamap-imdt-sbc8550.dtsi
+++ b/qcom/kalamap-imdt-sbc8550.dtsi
@@ -785,4 +785,14 @@
 
 		gpio-open-drain;
 	};
+	
+	/* Alternative firmware loader for the ADSP */
+	/* Audio Device Tree uses the adsp-loader */
+	ssc_sensors: qcom,msm-ssc-sensors {
+		compatible = "qcom,msm-ssc-sensors";
+		status = "okay";
+		qcom,firmware-name = "adsp";
+		qcom,rproc-handle = <&adsp_pas>;
+	};
+
 };
-- 
2.41.0

